### PR TITLE
RuntimeClass GA

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
Also fixed the version of PodOverhead feature flag

/sig node
/hold

Hold as waiting on k/k PR to be merged.

KEP: https://github.com/kubernetes/enhancements/issues/585
k/k: https://github.com/kubernetes/kubernetes/pull/95718

### CODE REVIEW

File: content/en/docs/setup/production-environment/windows/user-guide-windows-containers.md
Comment (SergeyKanzhelev) [inline content/en/docs/setup/production-environment/windows/user-guide-windows-containers.md:180]: there are many occurences of this in other language docs. Should I include those in this PR?

Comment (annajung) [reply content/en/docs/setup/production-environment/windows/user-guide-windows-containers.md:180]: pinging @sftim @kbhawkey @tengqm who can help answer this question

Comment (kbhawkey) [reply content/en/docs/setup/production-environment/windows/user-guide-windows-containers.md:180]: Hi. Typically, you should limit pull request changes to one language and let the localization teams update the
corresponding content.
I am unaware of a formal process for localization teams to follow-up on updates to the `dev-*` branch.
If the changes are straightforward, it would be a possible to get review/approval from a localization team before approving this PR.

Comment (SergeyKanzhelev) [reply content/en/docs/setup/production-environment/windows/user-guide-windows-containers.md:180]: I can send a separate PR than once this it merged

File: content/en/docs/concepts/containers/runtime-class.md
Comment (sftim) [inline content/en/docs/concepts/containers/runtime-class.md:12]:
```suggestion
{{< feature-state for_k8s_version="v1.20" state="stable" >}}
```

**Review by sftim — CHANGES_REQUESTED** [review]
The feature state specified here isn't valid; that's the only thing to fix that I can see.

File: content/en/docs/setup/production-environment/windows/user-guide-windows-containers.md
Comment (sftim) [reply content/en/docs/setup/production-environment/windows/user-guide-windows-containers.md:180]: _several_ separate PR's I'm afraid - each language has its own reviewers. Most of us can't review other localizations although we have a few polyglots.

File: content/en/docs/reference/access-authn-authz/admission-controllers.md
Comment (tallclair) [inline content/en/docs/reference/access-authn-authz/admission-controllers.md:728]: Shouldn't this be v1.20 stable? The admission controller is used for runtimeclass scheduling, not just pod overhead.

Comment (SergeyKanzhelev) [reply content/en/docs/reference/access-authn-authz/admission-controllers.md:728]: Yeah... I was thinking about it. Making it stable creates false impression that PodOverhead is also stable. @sftim what would you suggest? RuntimeClass feature is stable, PodOverhead which is part of the same API is in beta

File: content/en/docs/reference/access-authn-authz/admission-controllers.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/admission-controllers.md:731]: Is this accurate:
```suggestion
If you enable the `PodOverhead` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/), and define a RuntimeClass with [Pod overhead](/docs/concepts/scheduling-eviction/pod-overhead/) configured, this admission controller checks incoming
Pods. When enabled, this admission controller rejects any Pod create requests that have the overhead already set.
For Pods that have a  RuntimeClass is configured and selected in their `.spec`, this admission controller sets `.spec.overhead` in the Pod based on the value defined in the corresponding RuntimeClass.

{{< note >}}
The `.spec.overhead` field for Pod and the `.overhead` field for RuntimeClass are both in beta. If you do not enable the `PodOverhead` feature gate, all Pods are treated as if `.spec.overhead` is unset.
{{< /note >}}
```

**Review by sftim — COMMENTED** [review]
The gap in my knowledge is how Kubernetes behaves when a stable API such as `Pod` references an unstable feature such as `PodOverhead`.

I've guessed at an approach; if the behavior I'm describing matches my guess, then this is how I'd document it.

File: content/en/docs/reference/access-authn-authz/admission-controllers.md
Comment (SergeyKanzhelev) [reply content/en/docs/reference/access-authn-authz/admission-controllers.md:728]: This is exactly what I asked on the last SIG node meeting as I also lack context on how it is typically handled in k8s. So the outcome of the discussion was that this is fine. It is similar to the situation when the new feature being enabled via feature gate on the already stable API. So the text you suggested looks accurate

Comment (SergeyKanzhelev) [reply content/en/docs/reference/access-authn-authz/admission-controllers.md:728]:
```suggestion
{{< feature-state for_k8s_version="v1.20" state="stable" >}}
```

File: content/en/docs/reference/access-authn-authz/admission-controllers.md
Comment (SergeyKanzhelev) [inline content/en/docs/reference/access-authn-authz/admission-controllers.md:736]: Should I do this? Or it will be too much?

```suggestion
{{< note >}}

{{< feature-state for_k8s_version="v1.18" state="beta" >}}

The `.spec.overhead` field for Pod and the `.overhead` field for RuntimeClass are both in beta. If you do not enable the `PodOverhead` feature gate, all Pods are treated as if `.spec.overhead` is unset.
{{< /note >}}
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md
Comment (SergeyKanzhelev) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:130]:
```suggestion
| `PodOverhead` | `false` | Alpha | 1.16 | 1.17 |
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md
Comment (SergeyKanzhelev) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:131]:
```suggestion
| `PodOverhead` | `true` | Beta | 1.18 |  |
```

File: content/en/docs/reference/access-authn-authz/admission-controllers.md
Comment (tengqm) [reply content/en/docs/reference/access-authn-authz/admission-controllers.md:736]: looks like an overkill to me. RuntimeClass is GA but the `.overhead` field in it is beta. The `feature-state` line on 736 may cause some confusion. The text is already pretty clear.

**Review by irvifa — COMMENTED** [review]
/approve

**Comment by SergeyKanzhelev** [comment]
@irvifa there is already lgtm from @tallclair: https://github.com/kubernetes/website/pull/24698#issuecomment-720655396

**Comment by SergeyKanzhelev** [comment]
PR is on hold to have k/k PR merged first

**Review by irvifa — COMMENTED** [review]
/unhold

**Pull request merged**
